### PR TITLE
Add support for Interaction localization

### PR DESCRIPTION
**Pull request by DRSchlaubi**
https://discord.com/developers/docs/interactions/receiving-and-responding#interaction-object

### CODE REVIEW

**Review by HopeBaron — APPROVED** [review]

**Review by Lukellmann — APPROVED** [review]

**Comment by Lukellmann** [comment]
https://github.com/discord/discord-api-docs/commit/c39627c76f049c9f3c26fe40edde75017b76ba7e: would it make sense to
a) change `Locale` into a sealed class / interface with `object` subtypes for the listed locales and a `data class Unknown` subtype for unknown ones
b) use the `Locale` type in other places too (see Links in that commit) - probably something for another PR

**Comment by DRSchlaubi** [comment]
> [discord/discord-api-docs@c39627c](https://github.com/discord/discord-api-docs/commit/c39627c76f049c9f3c26fe40edde75017b76ba7e): would it make sense to a) change `Locale` into a sealed class / interface with `object` subtypes for the listed locales and a `data class Unknown` subtype for unknown ones b) use the `Locale` type in other places too (see Links in that commit) - probably something for another PR

Ofc they add this after I made this PR sigh

But yes I aggree we should add some logic for that, although I am uncertain whether we should use subtypes or static variables storing an instance for every language like `java.util.Locale`

Anyway as this would also affect different entities (as you mentioned previously) I would move that into another PR

**Comment by Lukellmann** [comment]
Yeah, it's probably better to just have some constants as static variables, they don't really have their own type after all.

File: common/src/main/kotlin/Locale.kt [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+         * Decodes the language from a `languageCode_countryCode` or `languageCode` format.
+         */
+        fun fromString(string: String): Locale {
+            val match = languageTagFormat.matchEntire(string) ?: error("$string is not a valid Locale")
+            val (language) = match.destructured
+            val country = match.groupValues[2]
+
+            return ALL.firstOrNull { (l, c) ->
+                language == l && country == (c ?: "")
+            } ?: error("This locale is not supported by Discord")
```
Comment (Lukellmann) [inline common/src/main/kotlin/Locale.kt:226]: Should this really throw an exception? There is still a public constructor.

Comment (DRSchlaubi) [reply common/src/main/kotlin/Locale.kt:226]: So you mean I should just default to Locale(country, language)?

Comment (Lukellmann) [reply common/src/main/kotlin/Locale.kt:226]: Yes, this way Discord randomly adding new locales won't cause exceptions.

Comment (HopeBaron) [reply common/src/main/kotlin/Locale.kt:226]: This is a common pattern in our `types` values, making the system error tolerant when Discord adds something to this.

**Pull request merged**